community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/planner/logical/cardinality/ExpressionSelectivityCalculator.scala

```diff
@@ -143,15 +148,18 @@ case class ExpressionSelectivityCalculator(stats: GraphStatistics, combiner: Sel
      * s in (0,1) = (1 - e) * f
      * return min(x, e + s in (0,1))
      */
-    val equality = java.math.BigDecimal.valueOf(calculateSelectivityForPropertyEquality(identifier, None, selections, propertyKey).factor)
-    val prefixLength = java.math.BigDecimal.valueOf(prefix.length)
-    val factor = java.math.BigDecimal.ONE.divide(prefixLength, 17, RoundingMode.HALF_UP)
-      .multiply(java.math.BigDecimal.valueOf(DEFAULT_RANGE_SEEK_FACTOR)).stripTrailingZeros()
+    val equality = math.BigDecimal.valueOf(calculateSelectivityForPropertyEquality(identifier, None, selections, propertyKey).factor)
+    val prefixLength = math.BigDecimal.valueOf(prefix match {
+      case Some(n) => n.length + 1
+      case None => DEFAULT_PREFIX_LENGTH
+    })
+    val factor = math.BigDecimal.ONE.divide(prefixLength, 17, RoundingMode.HALF_UP)
+      .multiply(math.BigDecimal.valueOf(DEFAULT_RANGE_SEEK_FACTOR)).stripTrailingZeros()
     val slack = BigDecimalCombiner.negate(equality).multiply(factor)
     val result = Selectivity(equality.add(slack).doubleValue())
```

Comment:
I believe you have a merge conflict around here. It should now be something like `Selectivity.of(...)` and then you can get rid of the `existence > Selectivity.ZERO` check since it should no longer be possible to create selectivities that are not in `[0, 1]`